code/controllers/subsystem/research.dm

```diff
@@ -293,3 +299,18 @@ SUBSYSTEM_DEF(research)
 			else
 				techweb_boost_items[path] = list(node.id = node.boost_item_paths[path])
 		CHECK_TICK
+
+/datum/controller/subsystem/research/proc/populate_ordnance_experiments()
+	ordnance_experiments = list()
+	scientific_partners = list()
+
+	for (var/experiment_path in subtypesof(/datum/experiment/ordnance))
+		var/datum/experiment/ordnance/typecasted_path = experiment_path
+		if (initial(typecasted_path.experiment_proper))
+			ordnance_experiments += new typecasted_path
```

Comment:
```suggestion
	for (var/datum/experiment/ordnance/experiment_path as anything in subtypesof(/datum/experiment/ordnance))
		if (initial(experiment_path.experiment_proper))
			ordnance_experiments += new experiment_path()
```
* The `as anything` keyword removes the type-checking and does what you want.
* A little `()` in `new` is rather common. No need to include it if you don't want, but I think that it helps with clarity as an initializer proc.